application/models/SurveysInGroup.php

```diff
@@ -73,22 +73,18 @@ public static function getMinimalPermissionRead()
      */
     public function hasPermission($sPermission, $sCRUD = 'read', $iUserID = null)
```

Comment:
As you are updating so much about this function, I think it would be very helpfull to comment what does this function do and also about the logic inside. To me it was confusing from the very begining how different permissions from different objects interact here.

This will need a bigger test, so this instructions will be usefull for testing it.

Reply:
This model are used only for survey right inside this group

Permission are call here :  https://github.com/LimeSurvey/LimeSurvey/blob/e54d2f532b363e474f679e6392e59da06efb8000/application/models/Survey.php#L2287